lighthouse-core/lib/traces/tracing-processor.js

```diff
@@ -246,6 +235,28 @@ class TraceProcessor {
   }
 
   /**
+   * Provides the top level events on the main thread.
+   * @param {!traceviewer.Model} model
+   * @param {{traceEvents: !Array<!Object>}} trace
+   * @param {number=} startTime Optional start time (in ms) of range of interest. Defaults to trace start.
+   * @param {number=} endTime Optional end time (in ms) of range of interest. Defaults to trace end.
+   * @return {!Array<{start: number, end: number}>}
+   */
+  static getMainThreadTopLevelEvents(model, trace, startTime = 0, endTime = Infinity) {
+    // Find the main thread via the first TracingStartedInPage event in the trace
+    const startEvent = trace.traceEvents.find(event => event.name === 'TracingStartedInPage');
+    const mainThread = TraceProcessor._findMainThreadFromIds(model, startEvent.pid, startEvent.tid);
+
+    // TODO(bckenny): filter for top level slices ourselves?
```

Comment:
can we fix our longstanding #489 while you're here?

```js
// See https://github.com/GoogleChrome/lighthouse/issues/489
const schedulableTaskTitle = 'TaskQueueManager::ProcessTaskFromWorkQueue';

const events = mainThread.sliceGroup.slices.filter(slice => {
  return slice.title === schedulableTaskTitle && 
      slice.end > startTime && 
      slice.start < endTime;
});
```

Just tested this guy out and he's good. This also matches what we see in https://codereview.chromium.org/2610183003/patch/140001/150003